.github/workflows/packages.yml

```diff
@@ -0,0 +1,102 @@
+name: Build all core packages
+
+on:
+  pull_request:
+    paths:
+      - '.github/workflows/packages.yml'
+      - 'include/**'
+      - 'package/**'
+      - 'toolchain/**'
+      - 'config/**'
+jobs:
+  build:
+    name: Build packages with external toolchain
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: False
+      matrix:
+        include:
+          - name: malta-be
+            target: malta
+            subtarget: be
+
+    container: registry.gitlab.com/openwrt/buildbot/buildworker-3.4.1
+
+    steps:
+      - name: Checkout master directory
+        uses: actions/checkout@v2
+        with:
+          path: openwrt
+
+      - name: Fix permission
+        run: |
+          chown -R buildbot:buildbot openwrt
+
+      - name: Initialization environment
+        run: |
+          TARGET=$(echo ${{ matrix.target }})
+          SUBTARGET=$(echo ${{ matrix.subtarget }})
+          echo "TARGET=$TARGET" >> "$GITHUB_ENV"
+          echo "SUBTARGET=$SUBTARGET" >> "$GITHUB_ENV"
+
+      - name: Parse toolchain file
+        working-directory: openwrt
+        run: |
+          TOOLCHAIN_FILE=$(curl "https://downloads.cdn.openwrt.org/snapshots/targets/${{ env.TARGET }}/${{ env.SUBTARGET }}/sha256sums" \
+            | sed -n -e 's/.*\(openwrt-toolchain.*\).tar.xz/\1/p')
+
+          echo "TOOLCHAIN_FILE=$TOOLCHAIN_FILE" >> "$GITHUB_ENV"
+
+      - name: Download external toolchain
+        shell: su buildbot -c "sh -e {0}"
+        working-directory: openwrt
+        run: |
+          wget -O - https://downloads.cdn.openwrt.org/snapshots/targets/${{ env.TARGET }}/${{ env.SUBTARGET }}/${TOOLCHAIN_FILE}.tar.xz \
+            | tar --xz -xf -
+
+      - name: Create configuration
+        shell: su buildbot -c "sh -e {0}"
+        working-directory: openwrt
+        run: |
+          echo CONFIG_ALL=y >> .config
```

Comment:
BTW this is how the config is seeded on our buildbots:

```yaml
   seedconfig: |-
      CONFIG_BUILDBOT=y
      CONFIG_DEVEL=y
      # CONFIG_CCACHE is not set
      CONFIG_KERNEL_KALLSYMS=y
````
so if the idea is to prevent build breakage on our buildbots, then I would do it in similar way.

Reply:
I would like to build all core kernel and user space packages here to detect if some pull request breaks building them. The build bots use a 2 stage setup, with the SDK, but that is more complicated.

Reply:
> The build bots use a 2 stage setup, with the SDK, but that is more complicated.